src/components/expression/expression-panel.js

```diff
@@ -272,8 +285,16 @@ genotet.ExpressionPanel.prototype.formatGeneInput = function(isGeneRegex,
       });
   } else {
     var inputWords = geneInput.split(',');
-    geneNames = inputWords.filter(function(word) {
-      return word in this.data.matrixInfo.allGeneNames;
+    var lowerGeneNames = {};
+    Object.keys(this.data.matrixInfo.allGeneNames).forEach(function(name) {
+      lowerGeneNames[name.toLowerCase()] = name;
+    });
+    geneNames = [];
+    inputWords.forEach(function(word) {
```

Comment:
I think using forEach() is better. Because we want to return an array of geneNames, which are case-sensitive. filter() is used to pick up items from inputWords, which are case-insensitive.